### ISSUE TITLE
[V2] MSVO has artifacts in non-occluded areas on cameras with a low FOV

### ISSUE THREAD

**Issue by Sebioff** [Unity-Technologies/PostProcessing#438]
![2018-01-26_20-59-38](https://user-images.githubusercontent.com/3110063/35458374-5c85f4a6-02dc-11e8-87b7-9c749a73ee57.png)

Repro project:
[DeferredTest.zip](https://github.com/Unity-Technologies/PostProcessing/files/1669048/DeferredTest.zip)

**Comment by Chman** [comment]
The current implementation of MSVO isn't meant to be used with orthographic cameras or FOV lower than 10 due to precision issues. If you really need a FOV that low you should use the other AO mode.

May I ask why you need such a low FOV?

**Comment by Sebioff** [comment]
> May I ask why you need such a low FOV?

Simply because that's what I'd like my game to look like :) Almost isometric.
FOV of 10 is not the look I want but since this MSVO is very nice I'll consider it. Was just hoping this is a bug or something that could be improved.

**Comment by Chman** [comment]
(is that for Parkitect?)

Be wary that ultra-low FOV cameras may also affect other features that require depth information so this artistic choice may come with unforeseen technical limitations :)

**Comment by Sebioff** [comment]
Yes, for Parkitect :) We're nearing the end of development, so there's not too much left ahead that might be causing technical limitations. Although... we're currently testing a switch to deferred rendering :D
All looking good though. AO is one of the biggest missing pieces.

**Comment by nicobella** [comment]
in the subject of MSVAO artifacts I would love to see any update, at least a "no, we will not fix it" answer is ok, for this issue that I've posted a month ago
https://github.com/Unity-Technologies/PostProcessing/issues/413

**Comment by artyom-zuev** [comment]
This issue also severely affects high-resolution rendering, not just low field of view cameras. Rendering the game at 4K for promotional shots makes MSVO almost useless due to its radius seemingly being tied to resolution (not world-space distance) and due to this stepping artifact appearing over every single surface.

**Comment by Sebioff** [comment]
Hi @Chman ,
we have increased our FOV to 12 since, which has reduced these artifacts - on lower resolutions they are not extremely noticeable anymore, but as others have said the look of the AO is very dependant on resolution and on higher resolutions the lines pop out again.

Is there anything else we can look at to improve this?

**Comment by candycat1992** [comment]
I finally solved the bug by using ARGBHalf format instead of ARGB32 to render the depth. It solved all the artifacts.

**Comment by Sebioff** [comment]
Oh nice! :) Could you give a hint where to change that?

**Comment by candycat1992** [comment]
Find the file MultiScaleVO.cs, modify line 189 to line 192 with (replace RHalf with RFloat):
`
AllocArray(cmd, ShaderIDs.TiledDepth1, MipLevel.L3, RenderTextureFormat.RFloat, true);
AllocArray(cmd, ShaderIDs.TiledDepth2, MipLevel.L4, RenderTextureFormat.RFloat, true);
AllocArray(cmd, ShaderIDs.TiledDepth3, MipLevel.L5, RenderTextureFormat.RFloat, true);
AllocArray(cmd, ShaderIDs.TiledDepth4, MipLevel.L6, RenderTextureFormat.RFloat, true);
`

**Comment by Sebioff** [comment]
Thank you! I can confirm that this completely removes the lines.
